paddle/fluid/operators/batch_norm_op.cu

```diff
@@ -395,20 +531,94 @@ class BatchNormGradKernel<platform::CUDADeviceContext, T>
+                                /*sizeInBytes=*/&workspace_size));
+
+          workspace_ptr = workspace_tensor.mutable_data(
+              ctx.GetPlace(), transformed_x.type(), workspace_size);
+
+          CUDNN_ENFORCE(platform::dynload::cudnnBatchNormalizationBackwardEx(
+              /*handle=*/dev_ctx.cudnn_handle(),
+              /*mode=*/mode_,
+              /*bnOps=*/CUDNN_BATCHNORM_OPS_BN,
+              /*alphaDataDiff=*/CudnnDataType<T>::kOne(),
+              /*betaDataDiff=*/CudnnDataType<T>::kZero(),
+              /*alphaParamDiff=*/CudnnDataType<T>::kOne(),
+              /*betaParamDiff=*/CudnnDataType<T>::kZero(),
+              /*xDesc=*/data_desc_,
+              /*xData=*/transformed_x.template data<T>(),
+              /*yDesc=*/nullptr,
+              /*yData=*/nullptr,
+              /*dyDesc=*/data_desc_,
+              /*dyData=*/transformed_d_y.template data<T>(),
+              /*dzDesc=*/nullptr,
+              /*dzData=*/nullptr,
+              /*dxDesc=*/data_desc_,
+              /*dxData=*/transformed_d_x.template mutable_data<T>(
+                  ctx.GetPlace()),
+              /*dBnScaleBiasDesc=*/bn_param_desc_,
+              /*bnScaleData=*/scale->template data<BatchNormParamType<T>>(),
+              /*bnBiasData=*/nullptr,
+              /*dBnScaleData=*/d_scale
+                  ->template mutable_data<BatchNormParamType<T>>(
+                      ctx.GetPlace()),
+              /*dBnBiasData=*/d_bias
+                  ->template mutable_data<BatchNormParamType<T>>(
+                      ctx.GetPlace()),
+              /*epsilon=*/epsilon,
+              /*savedMean=*/saved_mean_data,
+              /*savedInvVariance=*/saved_var_data,
+              /*activationDesc=*/nullptr,
+              /*workspace=*/workspace_ptr,
+              /*workSpaceSizeInBytes=*/workspace_size,
+              /*reserveSpace=*/reserve_space_tensor->template data<T>(),
+              /*reserveSpaceSizeInBytes=*/reserve_space_size));
+        }
+#endif
+        if (!called) {
+          CUDNN_ENFORCE(platform::dynload::cudnnBatchNormalizationBackward(
+              dev_ctx.cudnn_handle(), mode_, CudnnDataType<T>::kOne(),
+              CudnnDataType<T>::kZero(), CudnnDataType<T>::kOne(),
+              CudnnDataType<T>::kZero(), data_desc_,
+              transformed_x.template data<T>(), data_desc_,
+              transformed_d_y.template data<T>(), data_desc_,
+              transformed_d_x.template mutable_data<T>(ctx.GetPlace()),
+              bn_param_desc_, scale->template data<BatchNormParamType<T>>(),
+              d_scale->template mutable_data<BatchNormParamType<T>>(
+                  ctx.GetPlace()),
+              d_bias->template mutable_data<BatchNormParamType<T>>(
+                  ctx.GetPlace()),
+              epsilon, saved_mean_data, saved_var_data));
+        }
+
+        if (data_layout_str == "NHWC" && compute_format == "NCHW") {
```

Comment:
else?

Reply:
else do nothing. Because when data_layout_str == "NHWC" && compute_format == "NCHW", we transform the data_format before, after computing, we should revert the data_format

Reply:
else do nothing. Because when data_layout_str == "NHWC" && compute_format == "NCHW", we transform the data_format before, after computing, we should revert the data_format